src/sonic-ctrmgrd/ctrmgr/container_state

```diff
@@ -0,0 +1,317 @@
+    state_data.update(dict(tbl.get(feature)[1]))
+
+
+def read_field(feature, field, default):
+    tbl = swsscommon.Table(state_db, 'FEATURE')
+    data = dict(tbl.get(feature)[1])
+    return data[field] if field in data else default
+
+
+def check_unset_labels(feature, version):
+     check if this feature is in unset list or not 
+    tbl = swsscommon.Table(state_db, KUBE_LABEL_TABLE)
+    labels = dict(tbl.get(KUBE_LABEL_UNSET_KEY)[1])
+    return _get_version_key(feature, version) in labels
+
+
+def drop_label(feature, version):
+    Set/drop label as required
+        Update is done in state-db.
+        ctrmgrd sets it with kube API server as required
+    
+    tbl = swsscommon.Table(state_db, KUBE_LABEL_TABLE)
+    name = _get_version_key(feature, version)
+    tbl.set(KUBE_LABEL_UNSET_KEY, [ (name, "")])
+    tbl.hdel(KUBE_LABEL_SET_KEY, name)
+        
+
+def update_data(feature, data):
+    debug_msg("{}: {{ {}: {} }}".format(feature, field, val))
+    tbl = swsscommon.Table(state_db, "FEATURE")
+    tbl.set(feature, list(data.items()))
+"""
+
+def read_data(feature):
+    global state_db, set_owner, state_data
+
+    db = ConfigDBConnector()
+    db.connect(wait_for_init=True, retry_on=True)
+    tbl = db.get_table('FEATURE')
+
+    if ((feature in tbl) and (SET_OWNER in tbl[feature])):
+        set_owner = tbl[feature][SET_OWNER]
+
+    state_db = ConfigDBConnector()
+    state_db.db_connect("STATE_DB", wait_for_init=False, retry_on=True)
+    tbl = state_db.get_table('FEATURE')
+    if feature in tbl:
+        state_data.update(tbl[feature])
+
+
+def read_field(feature, field, default):
+    tbl = state_db.get_table('FEATURE')
+    if (feature in tbl) and (field in tbl[feature]):
+        return tbl[feature][field]
+    else:
+        return default
+
+def check_unset_labels(feature, version):
+    tbl = state_db.get_table(KUBE_LABEL_TABLE)
+    if KUBE_LABEL_UNSET_KEY in tbl:
+        labels = tbl[KUBE_LABEL_UNSET_KEY]
+    else:
+        labels = {}
+    return _get_version_key(feature, version) in labels
+
+
+def drop_label(feature, version):
+    """ Set/drop label as required
+        Update is done in state-db.
+        ctrmgrd sets it with kube API server as required
+    """
+    name = _get_version_key(feature, version)
+    state_db.mod_entry(KUBE_LABEL_TABLE, KUBE_LABEL_UNSET_KEY, { name: "" })
+
+
+def update_data(feature, data):
+    debug_msg("{}: {}".format(feature, str(data)))
+    state_db.mod_entry("FEATURE", feature, data) 
+
+    
+def get_docker_id():
+    cmd = 'cat /proc/self/cgroup | grep -e ":memory:" | rev | cut -f1 -d\'/\' | rev'
+    proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+    output = proc.communicate()[0].decode("utf-8")
+    return output.strip()[:12]
+
+
+def to_int(s):
+    try:
+        return(int(s))
+    except ValueError as ex:
+        syslog.syslog(syslog.LOG_ERR, "Non convertible err={}".format(str(ex)))
+        return 0
+
+
+def instance_lower(feature, version):
```

Comment:
Instead of checking instance_lower, assuming kube to kube via 2nd manifest, would it be possible to remove the container based on age of the associated daemonset? Maybe we could see which container is associated with the older daemonset, and remove that one.